File: Client/src/app/Models/Trains/Train.ts
export interface Train {
    id: number;
    name: string;
    whereGoes: string;
    whereFrom: string;
    departure: Date;
    arrival: Date;
    freePlaces: number;
    occcupiedPlaces: number;
    typesPlace: { [key: string]: number };
}

File: Server/DAL/Entities/Train.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Entities
{
    public class Train
    {
        public int Id { set; get; }
        public string Name { set; get; }
        public string WhereGoes { set; get; }
        public string WhereFrom { set; get; }
        public DateTime Departure { set; get; }
        public DateTime Arrival { set; get; }
        public int FreePlaces { set; get; }
        public int OccupiedPlaces { set; get; }
        public ICollection<RailwayCarriage> Carriages { get; set; }
    }
}
